jerry-core/ecma/base/ecma-gc.cpp

```diff
@@ -563,35 +563,17 @@ ecma_gc_run (void)
 } /* ecma_gc_run */
 
 /**
- * Try to free some memory (depending on severity).
+ * Try to free some memory.
  */
 void
-ecma_try_to_give_back_some_memory (mem_try_give_memory_back_severity_t severity) /**< severity of
-                                                                                  *   the request */
+ecma_try_to_give_back_some_memory ()
 {
-  if (severity == MEM_TRY_GIVE_MEMORY_BACK_SEVERITY_LOW)
+  /*
+   * If there is enough newly allocated objects since last GC, probably it is worthwhile to start GC now.
+   * Otherwise, probability to free sufficient space is considered to be low.
+   */
+  if (ecma_gc_new_objects_since_last_gc * CONFIG_ECMA_GC_NEW_OBJECTS_SHARE_TO_START_GC > ecma_gc_objects_number)
```

Comment:
This is a performance heuristic, so it may be that the condition is `false`, but there are some objects that can be freed. In this case, we can get false-positive out-of-memory.